### PR TITLE
feat: remove source link from table when deleting

### LINKED ISSUE #97: Remove link from Table when a CSV-Source is deleted
When a CSV-Source is deleted, the corresponding table should remove its link to the source and add an warning `schema:warning`

Follow up task to #88

### PR DESCRIPTION
**Pull request by lucafurrer**
Removes the link to the csvSource from the table when the source is deleted.

I did a refactoring to make the code testable and wrote a test

fixes #97

### CODE REVIEW

File: apis/core/lib/domain/csv-source/delete.ts [hunk 1]
```diff
@@ -1,40 +1,54 @@
...
 
-  const csvSource = await store.get(csvSourceTerm)
+  const csvSource = await store.get(resource)
   if (!csvSource) return
 
-  // TODO: Delete Tables or prevent it
-  // Find related tables
-  // Delete them
+  // Remove links from tables
+  const tables = await getLinkedTablesForSource(csvSource.term)
```
Comment (tpluscode) [inline apis/core/lib/domain/csv-source/delete.ts:41]: Is it necessary to await here?

```suggestion
  const tables = getLinkedTablesForSource(csvSource.term)
```

File: apis/core/lib/domain/csv-source/delete.ts [hunk 2]
```diff
@@ -1,40 +1,54 @@
...
+  for await (const table of tables) {
+    const tableGraph = await store.get(table)
+    tableGraph?.dataset.delete($rdf.quad(tableGraph.term, cc.csvSource, csvSource.term))
+  }
```
Comment (tpluscode) [inline apis/core/lib/domain/csv-source/delete.ts:45]: I would shift this into table itself. At least if it was assigning `undefined`

```suggestion
  for await (const tableId of tables) {
    const table = await store.getResource<Table>(tableId)
    if (table) {
      table.csvSource = undefined
    }
  }
```

Which also reveals that `Table#csvSource` is not typed to allow falsy values. Removing this link without adjust the interface risks undetected runtime errors

File: apis/core/lib/domain/csv-source/delete.ts (hunk 1 above)
Comment (lucafurrer) [reply apis/core/lib/domain/csv-source/delete.ts:41]: True, we have that in the `for await`

Comment (lucafurrer) [reply apis/core/lib/domain/csv-source/delete.ts]: Or not ` TypeError: tables is not async iterable`

**Review by martinmaillard — APPROVED** [review]

**Pull request merged**